bleak/backends/corebluetooth/CentralManagerDelegate.py

```diff
@@ -146,11 +147,11 @@ async def connect_(self, peripheral: CBPeripheral, timeout=10.0) -> bool:
         self._connection_state = CMDConnectionState.PENDING
         self.central_manager.connectPeripheral_options_(peripheral, None)
 
-        start = time.time()
-        while self._connection_state == CMDConnectionState.PENDING:
-            await asyncio.sleep(0)
-            if time.time() - start > timeout:
-                raise TimeoutError("Failed to connect in %d seconds" % timeout)
+        try:
+            await asyncio.wait_for(self._connection_state_changed.wait(), timeout=timeout)
+        except TimeoutError:
+            logger.debug(f"Connection timed out after {timeout} seconds.")
+            return False
```

Comment:
I think it would be better to propagate the exception here by calling `raise` instead of returning `False`. This way callers can know why the connection failed without having to parse the debug log.

Reply:
This makes sense to me.